### PR TITLE
feat(VFileInput): update to v3

### LINKED ISSUE #9349: [Feature Request] Auto truncate length for v-file-input
### Problem to solve
A fixed length of string is not always appropriate in a responsive context, the available width is generally not the same between mobile phones and a desktop.

### Proposed solution
When setting the *truncate-length* property of the *v-file-input* to "**auto**", the component dynamically adapts the length of the string by truncating it in the middle, respecting the following conditions:

1. The name is truncated only if the width of the text (name + size + others) is larger that the width of the input (when it overflows)
2. If 1. is fulfilled, the name is truncated so that the width of the text equals the width of the input
3. The input stays always one line height

I would even suggest to put this as the default value.

It could eventually be extended to other components if suitable.

I hope this is technically possible.

### PR DESCRIPTION
**Pull request by nekosaur**
Fixes #9349

## Motivation and Context

v3

## Markup:

<details>

```vue
<template>
  <v-app>
    <div class="pa-10">
      <h3>Variants</h3>
      <template v-for="variant in ['underlined', 'outlined', 'filled', 'contained', 'plain']" :key="variant">
        <v-file-input :label="variant" :variant="variant" class="ma-5" />
      </template>

      <h3>Multiple</h3>
      <v-file-input label="bar" multiple class="ma-5" />

      <h3>Multiple Counter</h3>
      <v-file-input label="bar" multiple counter class="ma-5" />

      <h3>Multiple Counter ShowSize</h3>
      <v-file-input label="bar" multiple counter show-size class="ma-5" />

      <h3>Not Clearable</h3>
      <v-file-input label="bar" :clearable="false" class="ma-5" />

      <h3>No prepend icon</h3>
      <v-file-input label="bar" :prepend-outer-icon="null" class="ma-5" />

      <h3>Programmatic</h3>
      <v-file-input label="bar" v-model="file" class="ma-5" />
      {{ file }}
      <v-btn @click="set">set</v-btn>
      <v-btn @click="clear">clear</v-btn>
    </div>
  </v-app>
</template>

<script>
  export default {
    name: 'Playground',
    data: () => ({
      model: 1,
      disabled: false,
      file: null,
      toSet: null,
    }),
    watch: {
      file (value) {
        value && value.length && (this.toSet = value)
      },
    },
    methods: {
      set () {
        this.file = this.toSet
      },
      clear () {
        this.file = []
      },
    },
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/build/webpack.dev.config.js [hunk 1]
```diff
@@ -68,10 +68,10 @@ module.exports = merge(baseWebpackConfig, {
     ]
   },
   devServer: {
-    static: resolve('../dev'),
+    // static: resolve('../dev'),
     host: process.env.HOST || 'localhost',
     port: process.env.PORT || '8080',
-    firewall: false,
+    // firewall: false,
```
Comment (KaelWD) [inline packages/vuetify/build/webpack.dev.config.js:74]: Update to the new webpack-dev-server schema

File: packages/vuetify/src/components/VFileInput/VFileInput.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      validator: (v: boolean | number) => {
+        return (
+          typeof v === 'boolean' ||
+          [1000, 1024].includes(v)
+        )
+      },
+    },
+    modelValue: {
+      type: Array as PropType<File[] | undefined>,
+      // TODO: This breaks types??
```
Comment (KaelWD) [inline packages/vuetify/src/components/VFileInput/VFileInput.tsx:53]: Probably need `(value: any) =>`

File: packages/vuetify/src/components/VFileInput/VFileInput.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  },
+
+  setup (props, { attrs, slots }) {
+    const { t } = useLocale()
+    const fileValue = useProxiedModel(props, 'modelValue')
+    const rootRef = ref()
+
+    watch(() => props.modelValue, value => {
+      if (rootRef.value?.inputRef.value && (value == null || value?.length === 0)) {
+        // hmm, inputRef gets unwrapped?
```
Comment (KaelWD) [inline packages/vuetify/src/components/VFileInput/VFileInput.tsx:76]: Yep, `ref()` unwraps nested refs.

File: packages/vuetify/src/components/VFileInput/VFileInput.tsx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+              <VBtn
+                icon={ props.prependOuterIcon }
+                variant="text"
+                tabindex="-1"
+                onClick={ () => inputRef.value?.click() }
+                disabled={ props.disabled }
+              />
+            ) : undefined,
+
+            default: ({ isDirty, isFocused, inputRef, props: slotProps }) => (
```
Comment (KaelWD) [inline packages/vuetify/src/components/VFileInput/VFileInput.tsx:123]: Need interfaces for these slots

File: packages/vuetify/src/util/helpers.ts [hunk 5]
```diff
@@ -468,3 +467,21 @@ export function findChildren (vnode?: VNodeChild): ComponentInternalInstance[] {
 
   return []
 }
+
+export function pick<
```
Comment (KaelWD) [inline packages/vuetify/src/util/helpers.ts:471]: Duplicate of `extract` on 185

Comment (johnleider) [reply packages/vuetify/src/util/helpers.ts:471]: We talked about `pick` having better typing.

File: packages/vuetify/build/webpack.dev.config.js (hunk 1 above)
Comment (johnleider) [reply packages/vuetify/build/webpack.dev.config.js:74]: I wasn't able to find any information regarding this in webpack-dev-server@3.x, only 4.0-rc. Could you clarify this?

**Pull request merged**